tests/unit/pypyr/stepsrunner_test.py

```diff
@@ -489,72 +531,60 @@ def test_run_pipeline_steps_swallow_sequence(mock_invoke_step, mock_module):
 
                     assert str(err_info.value) == "arb error here 6"
 
-    mock_logger_debug.assert_has_calls == [call('step1 is complex.'),
-                                           call('step2 is complex.'),
-                                           call('step3 is complex.'),
-                                           call('step4 is complex.'),
-                                           call('step5 is complex.'),
-                                           call('step6 is a simple string.')]
-
-    mock_logger_info.mock_calls == [
+    mock_logger_debug.assert_has_calls([
+        call('step1 is complex.'),
+        call('step2 is complex.'),
+        call('step3 is complex.'),
+        call('step4 is complex.'),
+        call('step5 is complex.'),
+        call('step6 is a simple string.')],
+        any_order=True
+    )
+
+    assert mock_logger_info.call_args_list == [
         call('step1 not running because run is False.'),
         call('step2 not running because skip is True.'),
         call('step5 not running because skip is True.')]
 
-    mock_logger_error.assert_called_once_with(
-        "step4 Ignoring error because swallow is True "
-        "for this step.\n"
-        "ValueError: arb error here 4")
+    assert mock_invoke_step.call_count == 3
+
+    assert mock_logger_error.call_args_list == [
+        call(
+            "step4 Ignoring error because swallow is True "
+            "for this step.\n"
+            "ValueError: arb error here 4"
+        ),
+        call("Error while running step step6"),
+    ]
 
     assert mock_invoke_step.call_count == 3
-    assert mock_invoke_step.mock_calls == [call(context={
```

Comment:
does not make sense without `DeepCopyMagicMock`. replaced with code below

Reply:
yes, if just testing for the last iteration's input it just about works, but this was sloppy on my part. . . thanks for catching it! 🙏